### PR TITLE
feat: support user and room search simultaneously

### PR DESCRIPTION
**Pull request by amitojsingh366**
- adds support for user and room search simultaneously to `kousa`
- adds support for user and room search simultaneously to `kebab`
- adds support for user and room search simultaneously to `kibbeh`

### CODE REVIEW

**Review by overlisted — APPROVED** [review]

File: kebab/src/websocket/wrapper.ts [hunk 1]
```diff
@@ -65,8 +64,15 @@ export const wrap = (connection: Connection) => ({
   /**
    * Allows you to call functions that return information about the ws state
    */
+
   query: {
-    search: (query: string): Promise<{ items: Array<Room | User> }> =>
+    search: (query: string): Promise<{
+      items: Array<User | Room>,
```
Comment (overlisted) [inline kebab/src/websocket/wrapper.ts:70]: why did you keep this

Comment (amitojsingh366) [reply kebab/src/websocket/wrapper.ts:70]: it was already items?

File: kousa/lib/broth/message/misc/search.ex [hunk 2]
```diff
@@ -31,17 +33,24 @@ defmodule Broth.Message.Misc.Search do
       # currently not enforced, but once we have real DisplayRoom and
       # DisplayUser schemas we'll make sure Search.search outputs those.
       field(:items, {:array, :map})
+      field(:rooms, {:array, :map})
+      field(:users, {:array, :map})
```
Comment (benawad) [inline kousa/lib/broth/message/misc/search.ex]: This should be like https://github.com/benawad/dogehouse/blob/staging/kousa/lib/broth/message/user/get_followers.ex#L27 for the rooms/users

File: kousa/lib/broth/message/misc/search.ex [hunk 3]
```diff
@@ -31,17 +33,24 @@ defmodule Broth.Message.Misc.Search do
...
   @impl true
   def execute(changeset, state) do
     case apply_action(changeset, :validate) do
       {:ok, %{query: query}} ->
-        {:reply, %Reply{items: Search.search(query), nextCursor: nil}, state}
+        rooms = Rooms.search_name(query)
+        users = Users.search_username(query)
+        items = Enum.concat(rooms, users)
+
+        {:reply, %Reply{items: items, rooms: rooms, users: users, nextCursor: nil}, state}
```
Comment (benawad) [inline kousa/lib/broth/message/misc/search.ex:53]: add a test case for rooms/users field

File: kibbeh/src/modules/search/SearchBarController.tsx [hunk 4]
```diff
@@ -39,6 +39,13 @@ export const SearchBarController: React.FC<SearchControllerProps> = ({}) => {
     [text]
   );
   const { push } = useRouter();
+  const [results, setResults] = useState<Array<Room | User>>([]);
+
+  useEffect(() => {
+    if (data?.rooms && data?.users) {
+      setResults([...data.rooms, ...data.users]);
+    }
+  }, [data]);
```
Comment (benawad) [inline kibbeh/src/modules/search/SearchBarController.tsx]: don't useEffect/useState, just stick it in a variable

**Review by benawad — CHANGES_REQUESTED** [review]

File: kibbeh/src/modules/search/SearchBarController.tsx (hunk 4 above)
Comment (amitojsingh366) [reply kibbeh/src/modules/search/SearchBarController.tsx]: but data comes from a hook right, so i need to use useEffect/useState

File: kousa/lib/broth/message/misc/search.ex (hunk 2 above)
Comment (amitojsingh366) [reply kousa/lib/broth/message/misc/search.ex]: done

File: kousa/lib/broth/message/misc/search.ex (hunk 3 above)
Comment (amitojsingh366) [reply kousa/lib/broth/message/misc/search.ex:53]: done

File: kibbeh/src/modules/search/SearchBarController.tsx (hunk 4 above)
Comment (amitojsingh366) [reply kibbeh/src/modules/search/SearchBarController.tsx]: ok nvm, i removed them

**Review by nadirabbas — APPROVED** [review]

**Pull request merged**